boards/nano33ble/src/test/linear_log_test.rs

```diff
@@ -0,0 +1,326 @@
+//! Tests the log storage interface in linear mode. For testing in circular mode, see
+//! log_test.rs.
+//!
+//! The testing framework creates a non-circular log storage interface in flash and performs a
+//! series of writes and syncs to ensure that the non-circular log properly denies overly-large
+//! writes once it is full. For testing all of the general capabilities of the log storage
+//! interface, see log_test.rs.
+//!
+//! To run the test, add the following line to the imix boot sequence:
+//! ```
+//!     test::linear_log_test::run(mux_alarm, dynamic_deferred_caller);
+//! ```
+//! and use the `USER` and `RESET` buttons to manually erase the log and reboot the imix,
+//! respectively.
+
+use capsules::log;
+use capsules::virtual_alarm::{MuxAlarm, VirtualMuxAlarm};
+use core::cell::Cell;
+use kernel::common::cells::{NumericCellExt, TakeCell};
+use kernel::common::dynamic_deferred_call::DynamicDeferredCall;
+use kernel::debug;
+use kernel::hil::flash;
+use kernel::hil::log::{LogRead, LogReadClient, LogWrite, LogWriteClient};
+use kernel::hil::time::{Alarm, AlarmClient};
+use kernel::static_init;
+use kernel::storage_volume;
+use kernel::ReturnCode;
+use nrf52840::{
+    nvmc::{NrfPage, Nvmc},
+    rtc::Rtc,
+};
+
+// Allocate 4 KiB volume for log storage (the nano33ble page size is 4 KiB).
+storage_volume!(LINEAR_TEST_LOG, 4);
```

Comment:
several of the tests are specifically designed to test writes across multiple pages, so it might be better if this was 8?

Reply:
Good point! The imix had page size 512 so 1 KiB was 2 pages. Since the nano33ble has page size 4096, I've allocated 8 KiB which is also 2 pages.